manifests/system.pp

```diff
@@ -0,0 +1,48 @@
+class puppet_metrics_collector::system (
+  String        $output_dir                    = '/opt/puppetlabs/puppet-metrics-collector',
+  Integer       $collection_frequency          = 5, #minutes
+  Integer       $polling_frequency_seconds     = 1,
+  Integer       $retention_days                = 90,
+  String        $system_metrics_ensure         = present,
+  Boolean       $symlink_puppet_metrics_collector = true,
+) {
+  $scripts_dir = "${output_dir}/scripts"
+  $bin_dir     = "${output_dir}/bin"
+
+  #assume if output is defined, all of the rest will be too as the init.pp must be in use
+  #and thus we don't need to redefine these
+  if ! defined(File[$output_dir]) {
+    file { [ $output_dir, $scripts_dir, $bin_dir]:
+      ensure => directory,
+    }
+
+    file { "${scripts_dir}/generate_system_metrics":
+      ensure => present,
+      mode   => '0755',
+      source => 'puppet:///modules/puppet_metrics_collector/generate_system_metrics'
+    }
```

Comment:
The `generate_system_metrics` file declaration should be outside of the `if ! defined` block since it is not declared anywhere else. This caused the `generate_system_metrics` to be missing on my system when both of the classes below were included in my test system.

```
  include puppet_metrics_collector
  include puppet_metrics_collector::system
```